### DISCUSSION TITLE
Code cleanup and development freeze

### DISCUSSION THREAD

**Discussion by pxeger** [Vyxal/Vyxal#130] [Rewrite] [answered]
I want to cleanup and refactor all of Vyxal's code to be more stylistic under [PEP 8](https://www.python.org/dev/peps/pep-0008/) and what-not. Currently, not only is the code syntactically all <s>ugly</s> *inconsistent with the standards*, but it's not very maintainable because of its structure (e.g. the interpreter core is all in one file)

However, to prevent massive merge conflicts, this would have to be done with a halt of all other development.

My suggestion is to warn people that, on some pre-determined date, they must have finished whatever they're working on, and not start working on something else (unless they're masochists and want to deal with those conflicts). I will then clean up the code (with modern auto-formatters and refactoring tooling it shouldn't be too big an undertaking), and everyone else can update their forks and continue working on stuff thence.

For long-running projects like the `e` branch, and [my idea to rework the compiler to use bytecode](https://github.com/orgs/Vyxal/teams/transpiler-design-team/discussions/1), (depending on their state when we reach that point) they should be refactored independently so they don't have to be finished by that point.

What are your thoughts?

**Comment by ysthakur** [comment]
Yes please. I propose we have different files for different categories of builtins and use proper functions instead of evaluated strings for commands.py, at the very least.

**Reply by pxeger** [reply]
If my bytecode compiling idea gets implemented, it will necessitate solving this issue too

**Comment by mathcat4** [comment]
Nice idea, not including comments, right?

**Reply by pxeger** [reply]
We can clean up the comments too, like remove typos and perhaps move them around on screen if necessary. And add more!

**Reply by mathcat4** [reply]
Yes, nice!

**Comment by ysthakur** [comment]
I'd also like to propose that we all format our code with the same formatter before committing any changes, preferably [black](https://black.readthedocs.io/en/stable/). black can also be used [online](https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4ARXAm5dAD2IimZxl1N_WlkPinBFoXIfdFTaTVkGVeHShArYj9yPlDvwBA7LhGo8BvRQqDilPtgsfdKl-ha7EFp0Ma6lY_06IceKiVsJ3BpoICJM9wU1VJLD7l3qd5xTmo78LqThf9uibGWcWCD16LBOn0JK8rhhx_Gf2ClySDJtvm7zQJ1Z-Ipmv9D7I_zhjztfi2UTVsJp7917XToHBm2EoNZqyE8homtGskFIiif5EZthHQvvOj8S2gJx8_t_UpWp1ScpIsD_Xq83LX-B956I_EBIeNoGwZZPFC5zAIoMeiaC1jU-sdOHVucLJM_x-jkzMvK8Utdfvp9MMvKyTfb_BZoe0-FAc2ZVlXEpwYgJVAGdCXv3lQT4bpTXyBwDrDVrUeJEg5cXH4TUTNf-yo029ofjTcZgdwbwkBGElHbHHsQNOhuA4R9GbE2Xx6TfVmH9I4AsqU3ohV7t3GkBwkM8XInLiVOQZ4p5yjM-SW4u3I6_BUS8o2djSZaPvzZPDScXVk1OXu3w0wV7DfrgiK_dpzHntoOvqSHrNLg-Ea6zvV6G2nil3QBTBPl5PDtMwDKchtvwmlhnbvTOrh53X9EnSe8QtRKbMLO4pxx4bAJX-hVCXl5OHpCGZLowD7JdKj1-NctJy9DL99yr-X6yu7KwGCYG7t3fm-lt7Lg_HS9xbBrWDVKBKwM2F7hmR1_n9RFjznRBHD3OpHKlgiWjbWJI0Q6GhXazSt_NVH1KtFiY_UPzzuchkeq2AcjGvQd3-ZPkoFJkNU1Xx7q1i62bM0OKwmDiCfvlesDkc1kV7ClOmjkkdP1i8SGDUJ7f5143JpeLx2AAAADWcEIXZR38NAABigXYCAAA3axtWbHEZ_sCAAAAAARZWg==).

**Reply by mathcat4** [reply]
Yes, black's great

**Comment by ysthakur** [comment]
Every (important) function and class should also have detailed docstrings to help maintainers.

**Comment by ysthakur** [comment]
[Strawpoll](https://strawpoll.com/974ygod5h) for when we freeze development if this is accepted. You can choose multiple answers

**Comment by UnrelatedString** [comment]
I'm not familiar enough with the branch/fork structure to say how much of a pain that might be to deal with, but I would make a cleanup a very high priority. Any branches with functionality meant to be incorporated into master should do so as soon as possible.

**Reply by ysthakur** [reply]
Yeah, it's going to be a problem updating the experimental branch especially.

**Comment by ysthakur** [comment]
[Project](https://github.com/Vyxal/Vyxal/projects/2) to plan this and flesh it out a bit more.

**Comment by lyxal** [comment]
Well then. We shall have a clean up.

**Comment by chunkybanana** [comment]
I would help but I think I'd mess it up more...

**Comment by pxeger** [comment, marked as answer]
This is irrelevant now, so I'm closing.
